### PR TITLE
Support rustc -Zsymbol-mangling-version=v0

### PR DESCRIPTION
**Pull request by cuviper**

### CODE REVIEW

**Comment by cuviper** [comment]
See also https://github.com/alexcrichton/rustc-demangle/pull/23

**Comment by luser** [comment]
Sorry for letting this sit for two weeks! https://github.com/luser/rustfilt/pull/11 just got submitted, the patch has very minor differences. Do you think any of them are relevant?

File: src/main.rs [hunk 1]
```diff
@@ -33,7 +33,7 @@ mod tests;
 
 lazy_static! {
     // NOTE: Use [[:alnum::]] instead of \w to only match ASCII word characters, not unicode
-    static ref MANGLED_NAME_PATTERN: Regex = Regex::new(r"_ZN[\$\._[:alnum:]]*").unwrap();
+    static ref MANGLED_NAME_PATTERN: Regex = Regex::new(r"_[RZ]N[\$\._[:alnum:]]*").unwrap();
```
Comment (jsgf) [inline src/main.rs]: `_R` symbols don't necessarily start with `_RN`.

Comment (cuviper) [reply src/main.rs]: I see, you're right -- for reference: https://github.com/rust-lang/rfcs/blob/master/text/2603-symbol-name-mangling-v2.md#syntax-of-mangled-names

**Comment by cuviper** [comment]
I'll yield to #11.

**Pull request closed without merging**